internal/aws/cloudwatch/handler/request_structuredlog_handler_test.go

```diff
@@ -0,0 +1,37 @@
+// Copyright 2020, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package handler
+
+import (
+	"net/http"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws/request"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestAddUserAgentCWAgent(t *testing.T) {
+	httpReq, _ := http.NewRequest("POST", "", nil)
```

Comment:
Please check the error returned here with the package `require` and use `http.NoBody` instead of `nil`.

There is a few code path optimisations to be had using `http.NoBody`